database/skylink.go

```diff
@@ -18,11 +19,12 @@ type AllowListedSkylink struct {
 // BlockedSkylink is a skylink blocked by an external request.
 type BlockedSkylink struct {
 	ID                primitive.ObjectID `bson:"_id,omitempty"`
-	Skylink           string             `bson:"skylink"`
+	Failed            bool               `bson:"failed"`
+	Hash              crypto.Hash        `bson:"hash"`
```

Comment:
I'm on the fence here. MongoDB stores this as binary data. That's not an issue for things like indices.  
It makes a mongo document look like this:
```
{
    "_id" : ObjectId("61b36f84a61138ae861723e6"),
    "skylink" : "[...obfuscated..]-2fH1N0f_dBb9wFFGDQ",
    "reporter" : {
        "name" : "f",
        "email" : "m",
        "other_contact" : "",
        "unauthenticated" : true
    },
    "reverted" : false,
    "reverted_tags" : null,
    "tags" : [ 
        "copyright"
    ],
    "timestamp_added" : ISODate("2021-12-10T15:17:24.840Z"),
    "timestamp_reverted" : Date(-62135596800000),
    "hash" : { "$binary" : "gRcFbhMRv9UPWH3Ux6ZFvd2M/tnx9TdH/3QW/cBRRg0=", "$type" : "00" }
}
```
I'm wondering whether it's better to store the hash as a string and convert all methods that expect `crypto.Hash` to expecting strings.

Reply:
Yeah that's a bit weird. Just implement the [Marshaler](https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#Marshaler) and [Unmarshaler](https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#Unmarshaler) interfaces to make it serialise it as a string. We already do that for the same interfaces in the `json` package to make it marshal as `hex`. You can just internally call `hash.String()` and `hash.LoadString` to achieve that quickly.

Reply:
ended up  being a bit more annoying but it works